docs/adr/stac-generation-ingestion-user-stories.md

```diff
@@ -0,0 +1,193 @@
+
+**Goal:** enumerate all common use-cases so that we can confidently design modular infrastructure that can serve all use-cases.
+
+## Use Case 1: DPS Job Outputs Cataloging in User-STAC
+
+### User Story 1A: Single DPS Job Output Cataloging
+
+**As a** researcher running a data processing job,  
+**I want to** have the outputs of my DPS job automatically cataloged in the user-STAC,  
+**So that** I can easily discover, access, visualize, and share my processed data products.
+
+**Acceptance Criteria:**
+
+- I can specify a collection ID (new or existing) for my job outputs
+- If I don't specify a collection, items are added to my default username-based collection
+- I control which outputs get indexed by generating a simple JSON with asset key/href pairs (e.g., `{"data": "output.tif", "log": "log.txt"}`)
+- I don't need to understand STAC metadata structure to catalog my outputs
+- I can easily create the required information from Python or R
+- The system automatically generates all required STAC metadata from my minimal input
+- Output files are accessible via STAC API within minutes of job completion
+- Proper attribution and provenance information is included in the metadata
+
+### User Story 1B: Batch DPS Job Output Cataloging
+
+**As a** researcher running multiple related DPS jobs,  
+**I want to** catalog outputs from thousands of related jobs under a single collection,  
+**So that** I can easily discover, access, visualize, and share my processed data products.
+
+**Acceptance Criteria:**
+
+- All jobs can reference the same collection ID
+- If no collection is specified, all items flow into my default username-based collection
+- I control which outputs get indexed by generating a simple JSON with asset key/href pairs for each job
+- The system handles all complex STAC metadata generation without requiring me to be a STAC expert
+- The system can handle high-volume concurrent cataloging requests
+- The collection maintains integrity across thousands of added items
+- I can track ingestion progress for the entire batch of jobs
+- I can query subsets of my collection based on job-specific metadata
+
+## Use Case 2: Existing File Cataloging in User-STAC
+
+### User Story 2A: Manual File List Cataloging
+
+**As a** user with existing data files,  
+**I want to** catalog a specific list of files in the user-STAC,  
+**So that** I can make my pre-existing data discoverable and accessible.
+
+**Acceptance Criteria:**
+
+- I can create a new collection for my existing files
+- If I don't specify a collection, items are added to my default username-based collection
+- I control which files get indexed by providing a simple JSON with asset key/href pairs
+- I don't need to understand complex STAC metadata structures
+- I can provide a list of file locations (S3 URIs, etc.)
+- The system automatically generates appropriate metadata for each file
+- I can review and modify the generated metadata if needed
```

Comment:
before or after the submission? should users be able to test the metadata generation first or will it be easy to batch update/override?

Reply:
For now I think just making it possible to delete and/or re-write the STAC metadata would be ok. It would be great if we could provide a test sandbox but that feels like a feature for later.